chia/full_node/full_node.py

```diff
@@ -127,57 +153,147 @@ def __init__(
+        self._timelord_lock = None
+        self.weight_proof_handler = None
+        self._blockchain_lock_queue = None
+        self._maybe_blockchain_lock_ultra_priority = None
+        self._maybe_blockchain_lock_high_priority = None
+        self._maybe_blockchain_lock_low_priority = None
+
+    @property
+    def block_store(self) -> BlockStore:
+        assert self._block_store is not None
+        return self._block_store
+
+    @property
+    def _blockchain_lock_ultra_priority(self) -> LockClient:
+        assert self._maybe_blockchain_lock_ultra_priority is not None
+        return self._maybe_blockchain_lock_ultra_priority
+
+    @property
+    def _blockchain_lock_high_priority(self) -> LockClient:
+        assert self._maybe_blockchain_lock_high_priority is not None
+        return self._maybe_blockchain_lock_high_priority
+
+    @property
+    def _blockchain_lock_low_priority(self) -> LockClient:
+        assert self._maybe_blockchain_lock_low_priority is not None
+        return self._maybe_blockchain_lock_low_priority
+
+    @property
+    def timelord_lock(self) -> asyncio.Lock:
+        assert self._timelord_lock is not None
+        return self._timelord_lock
+
+    @property
+    def mempool_manager(self) -> MempoolManager:
+        assert self._mempool_manager is not None
+        return self._mempool_manager
+
+    @property
+    def blockchain(self) -> Blockchain:
+        assert self._blockchain is not None
+        return self._blockchain
+
+    @property
+    def coin_store(self) -> CoinStore:
+        assert self._coin_store is not None
+        return self._coin_store
+
+    @property
+    def respond_transaction_semaphore(self) -> asyncio.Semaphore:
+        assert self._respond_transaction_semaphore is not None
+        return self._respond_transaction_semaphore
+
+    @property
+    def transaction_queue(self) -> asyncio.PriorityQueue[Tuple[int, TransactionQueueEntry]]:
+        assert self._transaction_queue is not None
+        return self._transaction_queue
+
+    @property
+    def db_wrapper(self) -> DBWrapper2:
+        assert self._db_wrapper is not None
+        return self._db_wrapper
+
+    @property
+    def hint_store(self) -> HintStore:
+        assert self._hint_store is not None
+        return self._hint_store
+
+    @property
+    def new_peak_sem(self) -> asyncio.Semaphore:
+        assert self._new_peak_sem is not None
+        return self._new_peak_sem
+
+    @property
+    def compact_vdf_sem(self) -> asyncio.Semaphore:
+        assert self._compact_vdf_sem is not None
+        return self._compact_vdf_sem
+
     def _set_state_changed_callback(self, callback: Callable[..., Any]) -> None:
         self.state_changed_callback = callback
 
     async def _start(self) -> None:
-        self.timelord_lock = asyncio.Lock()
-        self.compact_vdf_sem = asyncio.Semaphore(4)
+        self._timelord_lock = asyncio.Lock()
+        self._compact_vdf_sem = asyncio.Semaphore(4)
```

Comment:
Why not just move this into `__init__`, make them non-optional and get rid of the related properties? Same question for all the other non-db related ones below.

Reply:
Some asyncio stuff expects to be created inside an event loop.  I don't know how required it is vs. just a thing it warns when you don't do.  Mostly though I was targeting handling that separately, maybe.

Reply:
Here is a similar case where it seems to work, at least with `Lock`, https://github.com/Chia-Network/chia-blockchain/blob/main/chia/data_layer/data_layer.py#L86, couldn't find any for `Semaphore` though.. are there any warnings? Because if not, this `Optional` + `@property` thing is for no reason :/